syntax = "proto3";

service HelloWorldService {
  rpc hello(HelloRequest) returns (HelloResponse) {}
}

message HelloRequest {
  string firstname = 1;
  string lastname = 2;
}

message HelloResponse {
  string text = 1;
}
message ProductData {
  string productId = 1;
  string productName = 2;
  string productCategory = 3;
  string productAmount = 4;
  string productUnit = 5;
}

message WarehouseData {
  string warehouseID = 1;
  string warehouseName = 2;
  string timestamp = 3;
  string warehouseCountry = 4;
  string warehouseCity = 5;
  string address = 6;
  repeated ProductData productData = 7;
}

service WarehouseService {
  rpc GetWarehouseData (WarehouseRequest) returns (WarehouseData) {}
}

message WarehouseRequest {
  string warehouseID = 1;
  string warehouseName = 2;
  string timestamp = 3;
  string warehouseCountry = 4;
  string warehouseCity = 5;
  string address = 6;
  repeated ProductData productData = 7;
}
